llm/ext_server/server.cpp

```diff
@@ -3188,26 +3188,30 @@ int main(int argc, char **argv) {
                     prompt = "";
                 }
 
-                json image_data;
-                if (body.count("image_data") != 0) {
-                    image_data = body["image_data"];
-                }
-                else
+                // create and queue the task
+                json responses;
                 {
-                    image_data = "";
+                    const int id_task = llama.queue_tasks.get_new_id();
+                    llama.queue_results.add_waiting_task_id(id_task);
+                    llama.request_completion(id_task, {{"prompt", prompt}}, true, -1);
+
+                    // get the result
+                    task_result result = llama.queue_results.recv(id_task);
+                    llama.queue_results.remove_waiting_task_id(id_task);
+                    if (!result.error) {
```

Comment:
Let's do the error check and return here to get it out of the way, that way the success case can be de-indented